plutus-ledger/src/LedgerBytes.hs

```diff
@@ -35,30 +36,35 @@ import           Language.PlutusTx.Lift
 import qualified Language.PlutusTx.Prelude        as P
 import           Web.HttpApiData                  (FromHttpApiData (..), ToHttpApiData (..))
 
-fromHex :: BS.ByteString -> LedgerBytes
-fromHex = LedgerBytes . asBSLiteral
+fromHex :: BS.ByteString -> Either String LedgerBytes
+fromHex = fmap LedgerBytes . asBSLiteral
     where
 
-    handleChar :: Word8 -> Word8
+    handleChar :: Word8 -> Either String Word8
     handleChar x
-        | x >= 48 && x <= 57 = x - 48 -- hexits 0-9
-        | x >= 97 && x <= 102 = x - 87 -- hexits a-f
-        | x >= 65 && x <= 70 = x - 55 -- hexits A-F
-        | otherwise = error "not a hexit"
+        | x >= 48 && x <= 57 = Right (x - 48) -- hexits 0-9
+        | x >= 97 && x <= 102 = Right (x - 87) -- hexits a-f
+        | x >= 65 && x <= 70 = Right (x - 55) -- hexits A-F
```

Comment:
Could we get rid of these magic numbers somehow?  Something like
```
chr_a :: Word8
chr_a = 97
```
and so on?  That would take up a lot of space though.

Reply:
How would we all feel about:

``` haskell
    handleChar :: Word8 -> Either String Word8
    handleChar x
        | x >= ord '0' && x <= ord '9' = Right (x - ord '0') -- hexits 0-9
        | x >= ord 'a' && x <= ord 'f' = Right (x - ord 'a' + 10) -- hexits a-f
        | x >= ord 'A' && x <= ord 'F' = Right (x - ord 'A' + 10) -- hexits A-F
        | otherwise = Left ("not a hexit: '" <> show x <> "'")

```

Reply:
If that's what it's actually doing, that seems *way* clearer.

Reply:
I had to use `c2w` (char to word) in the end, instead of `ord`, but yeah, that's what it's doing all right. :-)

Reply:
Aha!  I didn't know about `c2w`.  In the main PLC parser I wrote my own version of that, so I can save a few characters now...